service_configuration_lib/spark_config.py

```diff
@@ -132,6 +133,26 @@ def get_aws_credentials(
             log.warning(
                 'Tried to assume role with web identity but something went wrong ',
             )
+    elif use_web_identity:
+        token_path = os.environ.get('AWS_WEB_IDENTITY_TOKEN_FILE')
+        role_arn = os.environ.get('AWS_ROLE_ARN')
+        if not token_path or not role_arn:
+            raise Exception('Expected AWS_WEB_IDENTITY_TOKEN_FILE and AWS_ROLE_ARN to be set.')
+        with open(token_path) as token_file:
+            token = token_file.read()
+        sts_client = boto3.client('sts')
+        timestamp = int(time.time())
+        session = sts_client.assume_role_with_web_identity(
+            RoleArn=role_arn,
+            RoleSessionName=f'{service}-session-{timestamp}',
+            WebIdentityToken=token,
+            DurationSeconds=session_duration,
+        )
+        return (
+            session['Credentials']['AccessKeyId'],
+            session['Credentials']['SecretAccessKey'],
+            session['Credentials']['SessionToken'],
+        )
```

Comment:
I don't know how to add comment outside the scope of changes, so adding it here.

Maybe also catch the exception of FileNotFound error if user gives assume-aws-role option to suggest them to aws-profile instead. Something we were discussing yesterday.

Reply:
I meant for this case: https://github.com/Yelp/service_configuration_lib/blob/0219e2fe0b4ef23d63ca2f5eb3df2a120446c500/service_configuration_lib/spark_config.py#L189